weed/shell/command_volume_tier_move.go

```diff
@@ -68,27 +71,57 @@ func (c *commandVolumeTierMove) Do(args []string, commandEnv *CommandEnv, writer
 		return fmt.Errorf("source tier %s is the same as target tier %s", fromDiskType, toDiskType)
 	}
 
+	volumeIdChan := make(chan needle.VolumeId)
+	wg := sync.WaitGroup{}
+	go c.doVolumeTierMove(&wg, commandEnv, writer, volumeIdChan, toDiskType, *applyChange)
+
+	err, volumeIds := c.collectAllVolumeIds(commandEnv, fromDiskType, toDiskType, *collectionPattern, *fullPercentage, *quietPeriod, *volumesPerStep)
+	if err != nil {
+		return err
+	}
+	for len(volumeIds) > 0 {
+		for _, vid := range volumeIds {
+			volumeIdChan <- vid
+			if *volumesPerStep == 0 {
+				wg.Wait()
+			}
+		}
+		if *volumesPerStep > 0 {
+			wg.Wait()
+			err, volumeIds = c.collectAllVolumeIds(commandEnv, fromDiskType, toDiskType, *collectionPattern, *fullPercentage, *quietPeriod, *volumesPerStep)
```

Comment:
The info collected here may be stale. Need to adjust the local data structure directly.

Reply:
@chrislusf Can you explain why I don't understand

Reply:
"collectAllVolumeIds" will read from the master, which will wait for the volume servers to report their volumes. If the volume servers are slow to report, the info will be stale. 

For example, in some cases the "collectAllVolumeIds" may get the same info as the last time. The same volume may be picked again. But actually the volume has already been moved.